iron-fish/ironfish
# executes evm transactions on blocks

## Summary

executes evm transactions in Blockchain.newBlock so that mining manager executes evm transactions during block template creation

executes evm transactions when adding blocks to chain

changes 'verifyTx' to use 'simulateTx' to avoid modifying state, updates 'simulateTx' to turn off gas fees

updates BlockchainDBTransaction to reset the stateManager stateRoot when aborting transactions

adds 'abort' method to IronfishStateManager to revert all checkpoints when db transactions abort

adds evm cli commands for testing sending transaction with evm description

## Testing Plan

manual testing:
1. set activation sequence for evm transactions on devnet
2. start devnet node
3. stop devnet node
4. use a spending key with evm:putAccount to initialize an an account with public funds
5. start devnet node with '--forceMining'
6. start miner
7. use evm:sendTransaction to create and submit a transaction with an evm description (use the spending key and account nonce from step 4)
8. wait for transaction to be mined
9. stop devnet node, use evm:getAccount to check account balances and verify transaction was executed

## Documentation

Does this change require any updates to the Iron Fish Docs (ex. [the RPC API
Reference](https://ironfish.network/docs/onboarding/rpc/chain))? If yes, link a
related documentation pull request for the website.

```
[ ] Yes
```

## Breaking Change

Is this a breaking change? If yes, add notes below on why this is breaking and label it with `breaking-change-rpc` or `breaking-change-sdk`.

```
[ ] Yes
```
